### PR TITLE
Issue 1470 paths windows

### PR DESCRIPTION
**Pull request by KeynesYouDigIt**
**Related Issue(s):**

- #1470 

**Description:**
billy boy dab

Python 3.13 nuances the use of `os.path.isabs` - https://docs.python.org/3/library/os.path.html#os.path.isabs

So I changed the test to reflect that

### TODO
- [ ] some tests still failing!

**PR Checklist:**

- [x] Pre-commit hooks and tests pass (run `scripts/test`)
- [x] Documentation has been updated to reflect changes, if applicable
- [x] This PR maintains or improves overall codebase code coverage.
- [x] Changes are added to the [CHANGELOG](https://github.com/stac-utils/pystac/blob/main/CHANGELOG.md). See [the docs](https://pystac.readthedocs.io/en/latest/contributing.html#changelog) for information about adding to the changelog.

### CODE REVIEW

File: tests/test_utils.py [hunk 1]
```diff
@@ -219,7 +219,10 @@ def test_is_absolute_href(self) -> None:
             ("item.json", False),
             ("./item.json", False),
             ("../item.json", False),
-            ("/item.json", True),
+            # https://docs.python.org/3/library/os.path.html#os.path.isabs
+            # Windows requires a drive name.
+            ("/item.json", os.name != "nt"),
+            ("d:/item.json", os.name == "nt"),
```
Comment (KeynesYouDigIt) [inline tests/test_utils.py:225]: Let me know if it's worth splitting the tests out into windows and non-windows (I saw `    @pytest.mark.skipif(os.name != "nt", reason="Windows only test"`  just below but this seemed fine)

**Comment by KeynesYouDigIt** [comment]
Apologies, back up and at it! CI should pass -> https://github.com/KeynesYouDigIt/pystac/actions/runs/11873921875

**Comment by KeynesYouDigIt** [comment]
OK! @gadomski , ready for a look now. This is a nuance that might relate to @jsignell 's work on https://github.com/stac-utils/pystac/pull/1472 - namely, os.path changed a bit in 3.13.

**Comment by KeynesYouDigIt** [comment]
CI passing example https://github.com/KeynesYouDigIt/pystac/actions/runs/11990954337

File: tests/test_catalog.py [hunk 2]
```diff
@@ -832,12 +832,17 @@ def test_generate_subcatalogs_works_for_subcatalogs_with_same_ids(self) -> None:
...
 
         catalog.normalize_hrefs("/")
+
         for item in catalog.get_items(recursive=True):
             item_parent = item.get_parent()
             assert item_parent is not None
             parent_href = item_parent.self_href
             path_to_parent, _ = os.path.split(parent_href)
-            subcats = [el for el in path_to_parent.split("/") if el]
+            # subcats = [el for el in path_to_parent.split("/") if el]
```
Comment (gadomski) [inline tests/test_catalog.py:841]:
```suggestion
```

File: tests/test_layout.py [hunk 3]
```diff
@@ -413,6 +415,15 @@ class AsIsLayoutStrategyTest(unittest.TestCase):
     def setUp(self) -> None:
         self.strategy = AsIsLayoutStrategy()
 
+        path_includes_drive: bool = (
+            sys.version_info.major == 3
+            and sys.version_info.minor >= 13
+            and os.name == "nt"
+        )
```
Comment (gadomski) [inline tests/test_layout.py:422]: Can you refactor this check out to a function, in case we want to re-use it elsewhere in tests? Thanks.

**Comment by gadomski** [comment]
> Happy to touch up style and/or squash commits, LMK!

Thanks! This is great. No need to touch up commits, we squash-and-merge here.

File: tests/test_layout.py (hunk 3 above)
Comment (KeynesYouDigIt) [reply tests/test_layout.py:422]: Indeed I can! should be able to do that very soon.

**Comment by KeynesYouDigIt** [comment]
@gadomski Let me know if this is what you had in mind, thanks!

**Review by gadomski — APPROVED** [review]

**Comment by gadomski** [comment]
Awesome, thank you! Looks great.

**Pull request merged**